Updated installation instructions

JeroenMols opened the pull request:
Website currently doesn't mention use of apt plugin, hence not providing a complete solution to get ButterKnife up and running.

Note: the apt-plugin version is static (but I saw the JS versioning isn't working on github pages)

jaredsburrows commented on website/index.html:
```diff
@@ -211,10 +211,28 @@ <h4 id="bonus">Bonus</h4>
+            <pre class="prettyprint">
+buildscript {
+  repositories {
+    mavenCentral()
```
For `buildscript` `repositories`, you do not need `mavenCentral()`. Also, with the android gradle plugin `2.2.0` out, you do not need `android-apt`. You can use the annotation processor that comes with it.

jaredsburrows commented on website/index.html:
```diff
@@ -211,10 +211,28 @@ <h4 id="bonus">Bonus</h4>
+apply plugin: 'android-apt'
 
+dependencies {
+  compile 'com.jakewharton:butterknife:<span class="version"><em>(insert latest version)</em></span>'
```
Put `8.4.0`, the latest version.

JeroenMols replied:
There is actually a piece of Javascript which loads the latest version automatically, but this is not working on GH pages, so hence better to add a fixed version.

JeroenMols commented:
Incorporated your feedback, Android plugin v2.2.0 stable is now available so made that the default installation instructions.

Also updated the README to use the built in annotation processor by default. 

Note that I removed the requirement to use Jack as this doesn't seem to be necessary. 
(tested with jackOptions.enabled false)

jaredsburrows commented:
@JeroenMols Looks great! I have a PR in parallel with this one: https://github.com/JakeWharton/butterknife/pull/749 that uses the latest plugin.

JakeWharton commented on README.md:
```diff
@@ -41,39 +41,26 @@ __Remember: A butter knife is like [a dagger][1] only infinitely less sharp.__
-    classpath 'com.neenbedankt.gradle.plugins:android-apt:1.8'
-  }
+    ...
+    dependencies {
```
use two-space indent

JakeWharton commented on README.md:
```diff
@@ -41,39 +41,26 @@ __Remember: A butter knife is like [a dagger][1] only infinitely less sharp.__
-  }
+    ...
+    dependencies {
+        classpath 'com.android.tools.build:gradle:2.2.0'  // 2.2.0 or higher
```
Comment is redundant with the sentence above and people will end up copy/pasting it into build files. Let's remove.

JakeWharton commented on website/index.html:
```diff
@@ -211,10 +211,25 @@ <h4 id="bonus">Bonus</h4>
+buildscript {
+    ...
+    dependencies {
+        classpath 'com.android.tools.build:gradle:2.2.0'  // 2.2.0 or higher
```
Ditto

JakeWharton commented on website/index.html:
```diff
@@ -211,10 +211,25 @@ <h4 id="bonus">Bonus</h4>
+            <pre class="prettyprint">
+dependencies {
+  compile 'com.jakewharton:butterknife:8.4.0</span>'
+  annotationProcessor 'com.jakewharton:butterknife-compiler:8.4.0</em></span>'
```
Both of these should use `<span class="version"><em>(latest version)</em></span>` for the version.

JeroenMols replied:
Added this change, works locally on my device, but not on Github pages it seems,
https://jakewharton.github.io/butterknife
![screen shot 2016-09-20 at 21 04 39](https://cloud.githubusercontent.com/assets/6319637/18684832/f22fa6f0-7f75-11e6-91e4-6a3df03c3cb9.png)

JakeWharton replied:
Probably needs to use https?

JeroenMols replied:
Only thing that isn't https is the ajax script:
<script src="http://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>

Shall I change that to https? (I'm not a web guru so just guessing...)

JakeWharton replied:
Someone fixed this on another project. I'll have to go track it down.

JeroenMols replied:
Side note: also okhttp, okio, retrofit, picasso and dagger have the same issue

JakeWharton commented on website/index.html:
```diff
@@ -211,10 +211,25 @@ <h4 id="bonus">Bonus</h4>
+            <pre class="prettyprint">
+buildscript {
+    ...
+    dependencies {
```
two space indent again

JeroenMols commented:
Incorporated review feedback

JakeWharton commented on website/index.html:
```diff
@@ -211,10 +211,25 @@ <h4 id="bonus">Bonus</h4>
+            <pre class="prettyprint">
+dependencies {
+  compile 'com.jakewharton:butterknife:<span class="version"><em>(latest version)</em></span></span>'
+  annotationProcessor 'com.jakewharton:butterknife-compiler:<span class="version"><em>(latest version)</em></span></em></span>'
```
You have an extra `</em></span>` here

JeroenMols commented:
Sorry for the late reply, didn't see I still had stuff to fix

tprochazka commented:
It was approved but not merged why? There is quite a lot open pull request without any explanation why it was not merged :-(

pumba210 commented:
Hi
